pkg/registry/batch/cronjob/strategy_test.go

```diff
@@ -428,3 +440,129 @@ func TestJobStrategy_WarningsOnUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestCronJobStrategy_Validate(t *testing.T) {
+	ctx := genericapirequest.NewDefaultContext()
+	if !Strategy.NamespaceScoped() {
+		t.Errorf("CronJob must be namespace scoped")
+	}
+	if Strategy.AllowCreateOnUpdate() {
+		t.Errorf("CronJob should not allow create on update")
+	}
+
+	var cronJobPodFailurePolicyIn0 = batch.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mycronjob",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: batch.CronJobSpec{
+			Schedule:          "* * * * ?",
+			ConcurrencyPolicy: batch.AllowConcurrent,
+			JobTemplate: batch.JobTemplateSpec{
+				Spec: batch.JobSpec{
+					Template: validPodTemplateSpecWithRestartPolicyNever,
+					PodFailurePolicy: &batch.PodFailurePolicy{
+						Rules: []batch.PodFailurePolicyRule{
+							{
+								Action: batch.PodFailurePolicyActionIgnore,
+								OnExitCodes: &batch.PodFailurePolicyOnExitCodesRequirement{
+									Operator: batch.PodFailurePolicyOnExitCodesOpIn,
+									Values:   []int32{0},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testcases := map[string]struct {
+		job *batch.CronJob
+		err *field.Error
+	}{
+		"job with invalid value 0 for In operator with PodFailurePolicy": {
```

Comment:
just one test case?

Reply:
Yes, cause currently we didn't change the validation logic for job create, we'll probably add some in next release.